modules/ocf_prometheus/files/gen-prometheus-nodes

```diff
@@ -2,32 +2,65 @@
 """Generate a list of hosts to give to prometheus for stats generation"""
 import json
 import os
+import sys
 import tempfile
-from sys import argv
+from argparse import ArgumentParser
 
 from ocflib.infra.hosts import hosts_by_filter
 
-hosts = []
-for vals in hosts_by_filter('(|(type=server)(type=desktop)(type=staffvm))'):
-    cn, ip, host_type = vals['cn'][0], vals['ipHostNumber'][0], vals['type']
-
-    hosts.append({
-        'targets': [ip + ':9100'],
-        'labels': {'host_type': host_type, 'instance': cn},
-    })
-
-if len(argv) < 2 or argv[1] == '-':
-    print(json.dumps(hosts, indent=4))
-else:
-    # atomic overwrite in case of errors
-    file_name = argv[1]
-    with tempfile.NamedTemporaryFile(mode='w', delete=False) as temp_file:
-        try:
-            json.dump(hosts, temp_file, indent=4)
-            os.chmod(temp_file.name, 0o644)
-            os.rename(temp_file.name, file_name)
-        finally:
+
+def get_hosts(types, port):
+    nodefilter = '(|{typefilters})'.format(
+        typefilters=''.join('(type={})'.format(nodetype) for nodetype in types)
+    )
+    hosts = []
+    for vals in hosts_by_filter(nodefilter):
+        cn, ip, host_type = vals['cn'][0], vals['ipHostNumber'][0], vals['type']
+
+        if port is not None:
+            ip = ip + ':{}'.format(port)
```

Comment:
Is this output useful at all without a port number? If not, we should require it and not bother with this edge case. My hunch is that it's better to be explicit about this anyways.

Reply:
It's useful because of our renumbering technique with the printers and switches.